desk/lib/channel-utils.hoon

```diff
@@ -613,144 +706,207 @@
 ::
 ++  simple-post-1
   |=  post=simple-post:c
-  ^-  simple-post:v7:old:c
-  %=  post
-    ::  seal
-    reply-meta  (reply-meta-1 reply-meta.post)
-    reacts      (reacts-1 reacts.post)
-    replies     (simple-replies-1 replies.post)
-    ::  essay
-    +  (essay-1 +.post)
+  ^-  simple-post:v7:c
+  :_  (essay-1 +.post)
+  ^-  simple-seal:v7:c
+  =,  -.post
+  :*  id
+      (reacts-1 reacts.post)
+      (simple-replies-1 replies.post)
+      (reply-meta-1 reply-meta.post)
   ==
 ::
 ++  simple-reply-1
   |=  =simple-reply:c
-  ^-  simple-reply:v7:old:c
+  ^-  simple-reply:v7:c
   %=  simple-reply
     +  (memo-1 +.simple-reply)
     reacts  (reacts-1 reacts.simple-reply)
   ==
 ::
 ++  simple-replies-1
   |=  replies=simple-replies:c
-  ^-  simple-replies:v7:old:c
+  ^-  simple-replies:v7:c
   %+  run:on-simple-replies:c  replies
   simple-reply-1
 ::
+++  reference-1
+  |=  ref=reference:c
+  ^-  reference:v7:c
+  ?-  -.ref
+      %post
+    =-  ref(post -)
+    ?:  ?=(%& -.post.ref)
+      (simple-post-1 +.post.ref)
+    %*  .  *simple-post:v7:c
+      content  `story:v7:c`[%inline ['[deleted message]']~]~
+      author  ~nul
+    ==
+  ::
+      %reply
+    =-  ref(reply -)
+    ?:  ?=(%& -.reply.ref)
+      (simple-reply-1 +.reply.ref)
+    %*  .  *simple-reply:v7:c
+      content  `story:v7:c`[%inline ['[deleted reply]']~]~
+      author   ~nul
+    ==
+  ==
+::
 ++  to-said-1
   |=  =said:c
-  ^-  said:v7:old:c
-  =-  said(q -)
-  ?-  -.q.said
-     %post  q.said(post (simple-post-1 post.q.said))
-     %reply  q.said(reply (simple-reply-1 reply.q.said))
-  ==
+  ^-  said:v7:c
+  ::NOTE  the tombstone branches here _shouldn't_ get hit in practice,
```

Comment:
Hm, does this mean ~nul posts above will actually never resolve?

Reply:
I think that's the intention